File: literary_life_backend/app/schemas/cycle.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import date, datetime


class CycleCreate(BaseModel):
    cycle_type: int = Field(default=7, ge=1, le=365)


class CycleResponse(BaseModel):
    id: int
    user_id: int
    cycle_type: int
    start_date: date
    end_date: date
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: literary_life_app/lib/models/cycle.dart
class WritingCycle {
  final int id;
  final int userId;
  final int cycleType;
  final DateTime startDate;
  final DateTime endDate;
  final String status;
  final DateTime createdAt;

  WritingCycle({
    required this.id,
    required this.userId,
    required this.cycleType,
    required this.startDate,
    required this.endDate,
    required this.status,
    required this.createdAt,
  });

  factory WritingCycle.fromJson(Map<String, dynamic> json) {
    return WritingCycle(
      id: json['id'],
      userId: json['user_id'],
      cycleType: json['cycle_type'],
      startDate: DateTime.parse(json['start_date']),
      endDate: DateTime.parse(json['end_date']),
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  int get daysRemaining {
    final now = DateTime.now();
    final pureEnd = DateTime(endDate.year, endDate.month, endDate.day);
    final pureNow = DateTime(now.year, now.month, now.day);
    final diff = pureEnd.difference(pureNow).inDays;
    return diff > 0 ? diff : 0;
  }

  bool get isActive => status == 'active';
}
